### PR TITLE
ast, executor: set Fields for SelectStmt in PrepareExec

### PR DESCRIPTION
**Pull request by XuHuaiyu**
PTAL @shenli

### CODE REVIEW

File: executor/adapter.go [hunk 1]
```diff
@@ -52,27 +53,33 @@ type recordSet struct {
...
-				DBName:       model.NewCIStr(dbName),
-				Table:        &model.TableInfo{Name: col.OrigTblName},
-				Column: &model.ColumnInfo{
-					FieldType: *col.RetType,
-					Name:      col.ColName,
-				},
-			}
-			a.fields = append(a.fields, rf)
-		}
+		return schema2ResultFields(a.executor.Schema(), a.stmt.Ctx.GetSessionVars().CurrentDB)
```
Comment (winkyao) [inline executor/adapter.go]: never set a.fields?

**Comment by XuHuaiyu** [comment]
PTAL @shenli @coocood @winkyao

File: ast/ast.go [hunk 2]
```diff
@@ -167,6 +167,9 @@ func RowToDatums(row types.Row, fields []*ResultField) []types.Datum {
 // Implementations include SelectStmt, SubqueryExpr, TableSource, TableName and Join.
 type ResultSetNode interface {
 	Node
+	// GetResultFields gets result fields.
+	// Will be deprecated soon. Do Not use it.
+	getResultFields() []*ResultField
```
Comment (coocood) [inline ast/ast.go]: Never used?

Comment (XuHuaiyu) [reply ast/ast.go]: yes, never used

File: executor/prepared_test.go [hunk 3]
```diff
@@ -113,6 +113,27 @@ func (s *testSuite) TestPrepared(c *C) {
 		_, err = tk.Exec("execute stmt")
 		c.Assert(err, NotNil)
 
+		_, _, fields, err := tk.Se.PrepareStmt("select a from prepare3")
```
Comment (shenli) [inline executor/prepared_test.go]: Could you add a test case with placeholder? And also a test case of update statement.

**Comment by coocood** [comment]
LGTM

**Pull request merged**